scroll-animations-1/Overview.bs

```diff
@@ -789,28 +770,66 @@ follows:
-    <div class="switch">
-    :   If [=effective start offset=] or [=effective end offset=] is null.
-    ::  The [=effective scroll range=] is null.
-
-    :   Otherwise
-    ::  The [=effective scroll range=] is the result of evaluating the following
-        expression:
-        <blockquote>
-        `[=effective end offset=] - [=effective start offset=]`
-        </blockquote>
-
-    </div>
-
+### The effective scroll offsets of a {{ScrollTimeline}} ### {#effective-scroll-offsets-algorithm}
+
+The <dfn>effective scroll offsets</dfn> are calculated by the procedure to
+<dfn>resolve scroll timeline offsets</dfn> from {{ScrollTimeline/scrollOffsets}}
+as follows:
+
+1.  Let <var>effective scroll offsets</var> be an empty list of
+    <a>effective scroll offset</a>s.
+1.  Let <var>first offset</var> be true.
+1.  If {{ScrollTimeline/scrollOffsets}} is empty
+    1.  Run the procedure to [=resolve a scroll timeline offset=] for 'auto'
+        with the <var>is first</var> flag set to <var>first offset</var> and
+        add the resulted value into <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+    1.  Run the procedure to [=resolve a scroll timeline offset=] for 'auto'
+        with the <var>is first</var> flag set to <var>first offset</var> and
+        add the resulted value into <var>effective scroll offsets</var>.
+1.  If {{ScrollTimeline/scrollOffsets}} has one element
+    1.  Run the procedure to [=resolve a scroll timeline offset=] for 'auto'
+        with the <var>is first</var> flag set to <var>first offset</var> and
+        add the resulted value into <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+1.  For each <var>scroll offset</var> in the list of
+    {{ScrollTimeline/scrollOffsets}}, perform the following steps:
+    1.  Let <var>effective offset</var> be a result of applying the procedure
+        to [=resolve a scroll timeline offset=] for <var>scroll offset</var>
+        with the <var>is first</var> flag set to <var>first offset</var>.
+    1.  If <var>effective offset</var> is null, the
+        [=effective scroll offsets=] is empty and abort the remaining steps.
+    1.  Add <var>effective offset</var> into
+        <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+1.  Return <var>effective scroll offsets</var>.
+
+### The progress of a {{ScrollTimeline}} ### {#progress-calculation-algorithm}
+The <dfn>scroll timeline progress</dfn> is calculated by applying the procedure
+to <dfn>calculate scroll timeline progress</dfn> given |offset| as follows:
+
+1.  Let <var>scroll offsets</var> be a result of applying the procedure to
+    [=resolve scroll timeline offsets=] for {{scrollOffsets}}.
+1.  Let <var>offset id</var> correspond to the position of the last offset in
+    <var>scroll offsets</var> whose value is less than or equal to
+    <var>offset</var> and the value at the following position greater than
```

Comment:
Does this mean that the specified scroll offsets have ordering constraints? If yes, what are they, and when/how are they enforced?

Reply:
There is no ordering constrains. Explanation why last interval is chosen is explained in the note below the section. I added another note explaining why matching interval will always be found.